### ISSUE TITLE
Retina userpics

### ISSUE THREAD

**Issue by nfagerlund** [dreamwidth/dw-free#2436]
Userpics: 100px x 100px. 👍🏼 

But a pixel isn't always a pixel! Web pixels were retconned into Device Independent Pixels, so each pixel of that 100x100 square ends up being four pixels on a retina screen. 

It'd be cool if userpics stayed the same _size_ but just looked a lot sharper and nicer, and these days that's pretty trivial on the frontend. 

So I propose allowing retina userpics. It'd work like this.

- Userpics have a density. To start with, we support 1x and 2x. (3x doesn't seem critical, but we should be able to enable new densities later.) 
- All existing userpics become 1x. 
- We quadruple the byte limit for individual userpics.
- When you upload a userpic, you can choose a density, defaulting to 1x. If you upload a userpic that's between 100px and 200px in at least one dimension and forget to set it to 2x, we ask whether you want to cancel or continue as 2x. (The degenerate case is where you post a 101x100 image, and you'd probably rather crop that than treat it as a 50x50 retina pic.)
- When you upload a pic at 2x density, we make a resized 1x version and store that too. 
- Anywhere we display userpics, we set the img's src to the 1x version, but use `<img srcset="...">` ([mdn](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Img#Using_the_srcset_attribute)) to offer the 2x version if it exists. Modern browsers grab whichever image they think is appropriate without downloading the other one, and hopefully userpics look nicer on dense displays. 
    - No exceptions for cases where the image is locked to 50x50 px, or whatever.
    - Possible global exception to be controlled by a cookie-based pref, so people with crap cell networks can trim their data usage a bit?

There might be some math about whether the increased resource usage pencils out? @zorkian's initial reaction was "yolo" (I'm paraphrasing), but also probably every RPer will immediately re-upload all 200 of their icons. My thinking before I'd asked anyone about this was to maybe treat 2x icons as two slots in your icon allotment if the accounting was tight, but man, probably much simpler to implement if we can just yolo.

**Comment by nfagerlund** [comment]
BTW, it looks like we already rigorously set height/width on userpic img tags, so the first thing I was worried about (image intended to be retina 70x70 expands to full 100x100 so now it's like 1.43x or something dumb) is already handled.

**Comment by rahaeli** [comment]
I am 100% for this as long as a) it doesn't risk icons trying to spread out of the 100x100 box (since everywhere and I do mean everywhere on the site assumes that) and b) is going to still look ok on non-retina displays! Which, from what I understand, should be the case, but I have a compulsive need to mention the immediate concerns, heh.

If Mark is fine with #yolo, I'm fine with #yolo. If it becomes a problem later we can revisit. I do really like the idea of a setting for users to reduce their data transfer if they have to, though, since I know it's a concern for some people and we frequently hear how much they appreciate our efforts to keep pages light. 

(Pie in the sky thought: if we could make it so that anyone using the Lynx skin (aka the skin that gets loaded if you reload in ?style=light) automatically gets the smaller version. But that's a nice-to-have, not a must or even a should.)

**Comment by kareila** [comment]
I know we hate to add more user settings, but maybe an option alongside image placeholders for people who want to save their bandwidth?

**Comment by rahaeli** [comment]
Yeah, that's what I was thinking -- I think this is worth the use of a setting.
